### ISSUE TITLE
Error: User cancelled image selection

### ISSUE THREAD

**Issue by umair-dpl** [ivpusic/react-native-image-crop-picker#1446]
### Version
Tell us which versions you are using: 

- react-native-image-crop-picker v0.?.?
- react-native v0.?.?

### Platform
Tell us to which platform this issue is related

- iOS
- Android

### Steps to reproduce

1.

2.

3.

### Attachments

// stacktrace or any other useful debug info

Love react-native-image-crop-picker? Please consider supporting our collective:
👉  https://opencollective.com/react-native-image-crop-picker/donate

**Comment by kutaisan** [comment]
Used like this and works like charm;

`try {
const image = await ImagePicker.openCropper({
          path: uri,
          height: 385,
          width: 1000,
        });
} catch(error){
console.log("Error:", error);
}`

**Comment by umair-dpl** [comment]
@kutaisan i am implementing it if it occurs again what is the fix then ?

**Comment by paruckerr** [comment]
inside catch(error)...

```javascript
if (error.code === 'E_PICKER_CANCELLED') {
  return false;
}
```

**Comment by RockDevelopr** [comment]
Same problem here. I tried all suggestions with no success.
Note: the problem occurs only in Android devices.

**Comment by ArchanaNair** [comment]
Getting similar issue on Andriod device. I am not getting any image as callback response, instead it is going to catch block. How to resolve this issue
 let img = await ImagePicker.openPicker({
                            width: 200,
                            height: 200,
                            compressImageMaxHeight: 400,
                            compressImageMaxWidth: 400,
                            includeBase64: true,
                            multiple: true,
                            mediaType: 'photo'
                        })
                        img.then(images => {                         
                          console.log( images)
                       
                        }).catch(error =>{
                            console.log("error in catch block", error.code)
                            if (error.code === 'E_PICKER_CANCELLED') {
                                return false;
                              }
                        })

**Comment by noumantahir** [comment]
Facing same issues, happening on android only, iOS is working fine

**Comment by sagar-vasu** [comment]
Facing same issues, happening on android device, emulator with Android 10 and 11 also working good, iOS is working fine...


Device: Mi Redmi Note 10
  Android 11
  
   ImagePicker.openPicker({
      mirrorImage: true,
      compressImageQuality: 0.5,
      forceJpg: true,
      writeTempFile: false,
      includeBase64: true,
    })
      .then(image => {
        alert(image);
        let ImageUri = Platform.OS === 'ios' ? image.sourceURL : image.path;
        setImage(ImageUri);
      })
      .catch(er => {
        alert(er);
        if (er.code === 'E_PICKER_CANCELLED') {
          // here the solution
          return false;
        }
      });

**Comment by ankitsingh3271** [comment]
> Facing same issues, happening on android device, emulator with Android 10 and 11 also working good, iOS is working fine...
> 
> Device: Mi Redmi Note 10
> Android 11
> 
> ImagePicker.openPicker({
> mirrorImage: true,
> compressImageQuality: 0.5,
> forceJpg: true,
> writeTempFile: false,
> includeBase64: true,
> })
> .then(image => {
> alert(image);
> let ImageUri = Platform.OS === 'ios' ? image.sourceURL : image.path;
> setImage(ImageUri);
> })
> .catch(er => {
> alert(er);
> if (er.code === 'E_PICKER_CANCELLED') {
> // here the solution
> return false;
> }
> });

This solution is working fine

**Comment by b3ni** [comment]
Same error

**Comment by ankitjaiswal1994** [comment]
@b3ni downgrading version to 0.35.3 worked for me. Not an ideal solution although.

**Comment by adhip94** [comment]
Yes, even i am facing the same issue on Android - Redmi. I try to select an image/video using `ImagePicker.openPicker` and it always goes to the catch block... 
Any alternatives for this?

**Comment by ErkanGorgulu** [comment]
I am having the same issue with my Xiaomi - Note 8 Pro. I choose an image from the gallery and it goes to catch with user canceled error. Downgrading to 0.35.3 didn't work. Taking images with camera works fine.

**Comment by mehtishamulhaq** [comment]
Facing the same issue on Redmi note 9. Is there any update regarding this ?

**Comment by davydspotnana** [comment]
Facing with the same issue `Xiaomi M2012K11AI, OS Level 30`

**Comment by davydspotnana** [comment]
I think it is not an issue of `react-native-image-crop-picker`,  you can fix it just:

```tsx
Just go to Settings -> Additional Settings -> Developer Option -> Turn on MIUI optimization (enable it)
Note - If not see the option "Turn on MIUI optimization" then click "Reset to default values" 4-5 times. It will enable the option.
```

If it not helps, try this:

```
this problem solved after i change android:launchMode="singleInstance" to singleTop in activity tag at androidManifest file but however i really dont know what are the differences between this two lunchMode in reactNative!!
```

**Comment by areebshakeel** [comment]
Facing same issues on Google Pixel 3 Android 12
ImagePicker.openPicker({
      mediaType: 'photo',
      forceJpg: true,
      multiple: false,
    })
.then(image => {
alert(image);
setImage(ImageUri);
})
.catch(er => {
alert(er);

});

**Comment by Pavanpuli89** [comment]
> I think it is not an issue of `react-native-image-crop-picker`, you can fix it just:
> 
> ```tsx
> Just go to Settings -> Additional Settings -> Developer Option -> Turn on MIUI optimization (enable it)
> Note - If not see the option "Turn on MIUI optimization" then click "Reset to default values" 4-5 times. It will enable the option.
> ```
> 
> If it not helps, try this:
> 
> ```
> this problem solved after i change android:launchMode="singleInstance" to singleTop 
> in activity tag at androidManifest file but however i really dont know 
> what are the differences between this two lunchMode in reactNative!!
> ```

This solution is working for MI mobile but facing the same issue in the google pixel 6 pro

**Comment by lchenfox** [comment]
The same issue on `XiaoMi 12 Pro Android 13`. I have already all permissions but it still returns 

```
{
    "nativeStackAndroid": [],
    "userInfo": null,
    "message": "User cancelled image selection",
    "code": "E_PICKER_CANCELLED"
}
```

How to fix this?

**Comment by lchenfox** [comment]
Hi, guys! After lots of tries. I solved this problem by adding `android:requestLegacyExternalStorage="true"` to AndroidManifest.xml like

```
    <application
      android:name=".MainApplication"
      android:label="@string/app_name"
      android:icon="@mipmap/ic_launcher"
      android:roundIcon="@mipmap/ic_launcher_round"
      android:allowBackup="false"
      android:requestLegacyExternalStorage="true" // Add this line.
      android:theme="@style/AppTheme">
```

I still don't know why. Hope it can help you.

**Comment by lchenfox** [comment]
> The same issue on `XiaoMi 12 Pro Android 13`. I have already granted all permissions but it still returns
> 
> ```
> {
>     "nativeStackAndroid": [],
>     "userInfo": null,
>     "message": "User cancelled image selection",
>     "code": "E_PICKER_CANCELLED"
> }
> ```
> 
> How to fix this?

Sorry for this. This issue happens because I call `` when I call `ImagePicker.openPicker` at the same time I call `await request(PERMISSIONS.ANDROID.READ_MEDIA_IMAGES)` as follows

```
const checkResult = await check(PERMISSIONS.ANDROID.READ_MEDIA_IMAGES);
if (checkResult === RESULTS.GRANTED) {
ImagePicker.openPicker.... // here I forgot to add "return".
}
const requestResult = await request(PERMISSIONS.ANDROID.READ_MEDIA_IMAGES);
```

OOPS! I made a rookie mistake... the fact that this issue has nothing to do with this repo.

**Comment by praveen-060522maticz** [comment]
> Just go to Settings -> Additional Settings -> Developer Option -> Turn on MIUI optimization (enable it)
> Note - If not see the option "Turn on MIUI optimization" then click "Reset to default values" 4-5 times. It will enable the option.

This is working on MI mobiles.
